### ISSUE TITLE
Can not execute tests on Android default Browser for real device

### ISSUE THREAD

**Issue by LimeDang** [appium/appium#4509]
Android real device 4.4.4 (Samsung device)
Appium 1.3.4

When execute with capability "browserName": "Browser" to test on Android default Browser the ChromeDriver log is: chrome is not reachable and Appium log is: Session can not create ...

Here is Appium log:
[36minfo[39m: Welcome to Appium v1.3.4 (REV c8c79a85fbd6870cd6fc3d66d038a115ebe22efe)
[36minfo[39m: Appium REST http interface listener started on 0.0.0.0:4723
[36minfo[39m: [debug] Non-default server args: {"udid":"7b70e0b7","fullReset":true,"defaultCommandTimeout":3600,"tmpDir":"/Users/admin/Appium_Temp"}
[36minfo[39m: Console LogLevel: debug
[36minfo[39m: [37m-->[39m [37mGET[39m [37m/wd/hub/status[39m [90m{}[39m
[36minfo[39m: [debug] Responding to client with success: {"status":0,"value":{"build":{"version":"1.3.4","revision":"c8c79a85fbd6870cd6fc3d66d038a115ebe22efe"}}}
[36minfo[39m: [37m<-- GET /wd/hub/status [39m[32m200[39m[90m 9.050 ms - 104[39m [90m{"status":0,"value":{"build":{"version":"1.3.4","revision":"c8c79a85fbd6870cd6fc3d66d038a115ebe22efe"}}}[39m
[36minfo[39m: [37m-->[39m [37mPOST[39m [37m/wd/hub/session[39m [90m{"desiredCapabilities":{"deviceName":"SM-G530H","platformName":"Android","browserName":"Browser"}}[39m
[36minfo[39m: Client User-Agent string: Apache-HttpClient/4.3.3 (java 1.5)
[36minfo[39m: Set mode: Proxying straight through to Chromedriver
[36minfo[39m: [debug] Looks like we want chrome on android
[36minfo[39m: [debug] Creating new appium session 42566fb2-6473-4b53-8529-4a75166ab52f
[36minfo[39m: [debug] Using fast reset? false
[36minfo[39m: [debug] Preparing device for session
[36minfo[39m: [debug] Not checking whether app is present since we are assuming it's already on the device
[36minfo[39m: [debug] Checking whether adb is present
[36minfo[39m: [debug] Using adb from /liemdang/Applications/android_sdk/platform-tools/adb
[36minfo[39m: Retrieving device
[36minfo[39m: [debug] Trying to find a connected android device
[36minfo[39m: [debug] Getting connected devices...
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb devices
[36minfo[39m: [debug] 2 device(s) connected
[36minfo[39m: Found device 7b70e0b7
[36minfo[39m: [debug] Setting device id to 7b70e0b7
[36minfo[39m: [debug] Waiting for device to be ready and to respond to shell commands (timeout = 5)
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 wait-for-device
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 shell "echo 'ready'"
[36minfo[39m: [debug] Starting logcat capture
[36minfo[39m: [debug] Pushing unlock helper app to device...
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 install "/usr/local/lib/node_modules/appium/build/unlock_apk/unlock_apk-debug.apk"
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 shell "dumpsys window"
[36minfo[39m: [debug] Screen already unlocked, continuing.
[36minfo[39m: [debug] Forwarding system:4724 to device:4724
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 forward tcp:4724 tcp:4724
[36minfo[39m: [debug] Pushing appium bootstrap to device...
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 push "/usr/local/lib/node_modules/appium/build/android_bootstrap/AppiumBootstrap.jar" /data/local/tmp/
[36minfo[39m: Starting App
[36minfo[39m: [debug] Attempting to kill all 'uiautomator' processes
[36minfo[39m: [debug] Getting all processes with 'uiautomator'
[36minfo[39m: [debug] executing cmd: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 shell "ps 'uiautomator'"
[36minfo[39m: [debug] No matching processes found
[36minfo[39m: [debug] Running bootstrap
[36minfo[39m: [debug] spawning: /liemdang/Applications/android_sdk/platform-tools/adb -s 7b70e0b7 shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: numtests=1[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: stream=[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] io.appium.android.bootstrap.Bootstrap:[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: id=UiAutomatorTestRunner[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: test=testRunServer[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: class=io.appium.android.bootstrap.Bootstrap[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: current=1[39m
[36minfo[39m: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS_CODE: 1[39m
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Socket opened on port 4724
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Appium Socket Server Ready
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Loading json...
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Registered crash watchers.
[36minfo[39m: [debug] Pushing command to appium work queue: ["getDataDir",{}]
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Client connected
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"getDataDir","params":{}}
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Got command action: getDataDir
[36minfo[39m: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"\/data\/local\/tmp","status":0}
[36minfo[39m: [debug] dataDir set to: /data/local/tmp
[36minfo[39m: [debug] Creating Chrome session
[36minfo[39m: [debug] Set chromedriver binary as: /usr/local/lib/node_modules/appium/build/chromedriver/mac/chromedriver
[36minfo[39m: [debug] Ensuring Chromedriver exists
[36minfo[39m: [debug] Killing any old chromedrivers, running: ps -ef | grep /usr/local/lib/node_modules/appium/build/chromedriver/mac/chromedriver | grep -v grep |grep -e '--port=9515\(\s.*\)\?$' | awk '{ print $2 }' | xargs kill -15
[36minfo[39m: [debug] Successfully cleaned up old chromedrivers
[36minfo[39m: [debug] Spawning chromedriver with: /usr/local/lib/node_modules/appium/build/chromedriver/mac/chromedriver
[36minfo[39m: [debug] [CHROMEDRIVER] Starting ChromeDriver (v2.10.267517) on port 9515
Only local connections are allowed.
[36minfo[39m: [debug] Making http request with opts: {"url":"http://127.0.0.1:9515/wd/hub/session","method":"POST","json":{"sessionId":null,"desiredCapabilities":{"chromeOptions":{"androidPackage":"com.android.browser","androidActivity":"com.android.browser.BrowserActivity","androidDeviceSerial":"7b70e0b7"}}}}
[36minfo[39m: [debug] Cleaning up appium session
[36minfo[39m: [debug] Error: Did not get session redirect from Chromedriver
    at null.<anonymous> (/usr/local/lib/node_modules/appium/lib/devices/android/chromedriver.js:222:12)
    at Request._callback (/usr/local/lib/node_modules/appium/lib/devices/common.js:121:5)
    at Request.self.callback (/usr/local/lib/node_modules/appium/node_modules/request/request.js:121:22)
    at Request.emit (events.js:98:17)
    at Request.<anonymous> (/usr/local/lib/node_modules/appium/node_modules/request/request.js:985:14)
    at Request.emit (events.js:117:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/appium/node_modules/request/request.js:936:12)
    at IncomingMessage.emit (events.js:117:20)
    at _stream_readable.js:944:16
    at process._tickDomainCallback (node.js:486:13)
[36minfo[39m: [debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: Did not get session redirect from Chromedriver)","origValue":"Did not get session redirect from Chromedriver"},"sessionId":null}
[36minfo[39m: [37m<-- POST /wd/hub/session [39m[31m500[39m[90m 66590.961 ms - 214[39m [90m[39m
 

Executing tests on Android default Browser for real device is not supported right?

Do you have any ideas or plans to support this?

**Comment by p00j4** [comment]
@LimeDang : does it works well for you on simulator !

**Comment by LimeDang** [comment]
It can work on Android Emulator.

**Comment by 0x1mason** [comment]
It is supported.  Try the latest appium release, 1.3.6.  It should have some help for this.

**Comment by LimeDang** [comment]
Try Appium 1.3.6 with Android device 4.4.4 (Samsung device) -> got the same Error: Did not get session redirect from Chromedriver.

**Comment by 0x1mason** [comment]
Does "Chrome" work?

**Comment by LimeDang** [comment]
Yes, Chrome works normally !

**Comment by 0x1mason** [comment]
Cool. We need to fix "Browser" tho. Thanks for the heads up and hope that
gets you unblocked.

**Comment by vir513** [comment]
Hi LimeDang,
                       I have tried chrome browser that works normally but when i try "browserName": "Browser" in emulator , it doesnot work. 
Are you using the same chrome driver to test on both chrome and default browser in an emulator??
My chromedriver version is 2.3

**Comment by vir513** [comment]
some information on the emulator that i am using, its Gennymotion

**Comment by tarunsujit** [comment]
@vir513 I tried in emulator (using Gennymotion) "browserName": "Browser" It worked fine with latest pull.
What version of Appium are you trying?

**Comment by vir513** [comment]
@tarunsujit my Appium version is 1.3.4.1

Error log in Appium:
> Starting Node Server
> info: Welcome to Appium v1.3.4 (REV c8c79a85fbd6870cd6fc3d66d038a115ebe22efe)
> info: Appium REST http interface listener started on 127.0.0.1:5000
> info: [debug] Non-default server args: {"address":"127.0.0.1","port":5000,"logNoColors":true,"platformName":"Android","platformVersion":"19","automationName":"Appium"}
> info: Console LogLevel: debug
> info: [37m-->[39m [37mPOST[39m [37m/wd/hub/session[39m [90m{"desiredCapabilities":{"platformVersion":"4.4.4","browserName":"Browser","platformName":"Android","deviceName":"Google Nexus 7"}}[39m
> info: Set mode: Proxying straight through to Chromedriver
> info: [debug] Looks like we want chrome on android
> info: [debug] Creating new appium session 2b7fac75-f457-43dd-83e8-30b8dd45af81
> info: [debug] Preparing device for session
> info: [debug] Not checking whether app is present since we are assuming it's already on the device
> info: [debug] Checking whether adb is present
> info: [debug] Using adb from C:\ADT\sdk\platform-tools\adb.exe
> info: Retrieving device
> info: [debug] Trying to find a connected android device
> info: [debug] Getting connected devices...
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" devices
> info: [debug] 1 device(s) connected
> info: Found device 192.168.56.101:5555
> info: [debug] Setting device id to 192.168.56.101:5555
> info: [debug] Waiting for device to be ready and to respond to shell commands (timeout = 5)
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 wait-for-device
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 shell "echo 'ready'"
> info: [debug] Starting logcat capture
> info: [debug] Pushing unlock helper app to device...
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 install "C:\Users\birmanga\Desktop\Appium\node_modules\appium\build\unlock_apk\unlock_apk-debug.apk"
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 shell "dumpsys window"
> info: [debug] Writing dumpsys output to C:\Users\birmanga\Desktop\Appium\node_modules\appium\.dumpsys.log
> info: [debug] Screen already unlocked, continuing.
> info: [debug] Forwarding system:4724 to device:4724
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 forward tcp:4724 tcp:4724
> info: [debug] Pushing appium bootstrap to device...
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 push "C:\\Users\\birmanga\\Desktop\\Appium\\node_modules\\appium\\build\\android_bootstrap\\AppiumBootstrap.jar" /data/local/tmp/
> info: Starting App
> info: [debug] Attempting to kill all 'uiautomator' processes
> info: [debug] Getting all processes with 'uiautomator'
> info: [debug] executing: "C:\ADT\sdk\platform-tools\adb.exe" -s 192.168.56.101:5555 shell "ps 'uiautomator'"
> info: [debug] No matching processes found
> info: [debug] Running bootstrap
> info: [debug] spawning: C:\ADT\sdk\platform-tools\adb.exe -s 192.168.56.101:5555 shell uiautomator runtest AppiumBootstrap.jar -c io.appium.android.bootstrap.Bootstrap
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: numtests=1[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: stream=[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] io.appium.android.bootstrap.Bootstrap:[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: id=UiAutomatorTestRunner[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: test=testRunServer[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: class=io.appium.android.bootstrap.Bootstrap[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS: current=1[39m
> info: [debug] [90m[UIAUTOMATOR STDOUT] INSTRUMENTATION_STATUS_CODE: 1[39m
> info: [debug] [BOOTSTRAP] [debug] Socket opened on port 4724
> info: [debug] [BOOTSTRAP] [debug] Appium Socket Server Ready
> info: [debug] Pushing command to appium work queue: ["getDataDir",{}]
> info: [debug] [BOOTSTRAP] [debug] Loading json...
> info: [debug] [BOOTSTRAP] [debug] Registered crash watchers.
> info: [debug] [BOOTSTRAP] [debug] Client connected
> info: [debug] dataDir set to: /data
> info: [debug] Creating Chrome session
> info: [debug] Set chromedriver binary as: C:\Users\birmanga\Desktop\Appium\node_modules\appium\build\chromedriver\windows\chromedriver.exe
> info: [debug] [BOOTSTRAP] [debug] Got data from client: {"cmd":"action","action":"getDataDir","params":{}}
> info: [debug] [BOOTSTRAP] [debug] Got command of type ACTION
> info: [debug] [BOOTSTRAP] [debug] Got command action: getDataDir
> info: [debug] [BOOTSTRAP] [debug] Returning result: {"value":"\/data","status":0}
> info: [debug] Ensuring Chromedriver exists
> info: [debug] Killing any old chromedrivers, running: FOR /F "usebackq tokens=5" %a in (`netstat -nao ^| findstr /R /C:"9515 "`) do (FOR /F "usebackq" %b in (`TASKLIST /FI "PID eq %a" ^| findstr /I chromedriver.exe`) do (IF NOT %b=="" TASKKILL /F /PID %b))
> info: [debug] No old chromedrivers seemed to exist
> info: [debug] Spawning chromedriver with: C:\Users\birmanga\Desktop\Appium\node_modules\appium\build\chromedriver\windows\chromedriver.exe
> info: [debug] [CHROMEDRIVER] Starting ChromeDriver (v2.3) on port 9515
> info: [debug] Making http request with opts: {"url":"http://127.0.0.1:9515/wd/hub/session","method":"POST","json":{"sessionId":null,"desiredCapabilities":{"chromeOptions":{"androidPackage":"com.android.browser","androidActivity":"com.android.browser.BrowserActivity","androidDeviceSerial":"192.168.56.101:5555"}}}}
> error: Chromedriver create session did not work. Status was 200 and body was {"sessionId":"8f57a55425c85963510112f2cbde63e4","status":13,"value":{"message":"unknown error: Chrome version must be >= 28.0.1475.0\n  (Driver info: chromedriver=2.3,platform=Windows NT 6.1 SP1 x86_64)"}}
> info: [debug] Cleaning up appium session
> info: [debug] Error: Did not get session redirect from Chromedriver
>     at null.<anonymous> (C:\Users\birmanga\Desktop\Appium\node_modules\appium\lib\devices\android\chromedriver.js:220:12)
>     at Request._callback (C:\Users\birmanga\Desktop\Appium\node_modules\appium\lib\devices\common.js:116:5)
>     at Request.self.callback (C:\Users\birmanga\Desktop\Appium\node_modules\appium\node_modules\request\request.js:121:22)
>     at Request.EventEmitter.emit (events.js:98:17)
>     at Request.<anonymous> (C:\Users\birmanga\Desktop\Appium\node_modules\appium\node_modules\request\request.js:985:14)
>     at Request.EventEmitter.emit (events.js:117:20)
>     at IncomingMessage.<anonymous> (C:\Users\birmanga\Desktop\Appium\node_modules\appium\node_modules\request\request.js:936:12)
>     at IncomingMessage.EventEmitter.emit (events.js:117:20)
>     at _stream_readable.js:920:16
>     at process._tickDomainCallback (node.js:459:13)
> info: [debug] Responding to client with error: {"status":33,"value":{"message":"A new session could not be created. (Original error: Did not get session redirect from Chromedriver)","origValue":"Did not get session redirect from Chromedriver"},"sessionId":null}
> error: Failed to start an Appium session, err was: Error: Did not get session redirect from Chromedriver
> info: [37m<-- POST /wd/hub/session [39m[31m500[39m[90m 4231.561 ms - 214[39m [90m[39m

My Desired Capabilities
DesiredCapabilities capabilities = new DesiredCapabilities();
			capabilities.setCapability("deviceName", "Google Nexus 7");//change device name(samsung-sm_n900) as per AVD name configured
			capabilities.setCapability("platformName", "Android");
			capabilities.setCapability("platformVersion", "4.4.4");
			capabilities.setCapability("browserName", "Browser");
			
			driver = new RemoteWebDriver(new URL("http://127.0.0.1:5000/wd/hub"),capabilities);
			ATUReports.setWebDriver(driver);

**Comment by 0x1mason** [comment]
If you're using the windows gui, you can cd to the Appium root directory and run `npm install appium`. Backup first. Then try with 1.3.6.

**Comment by vir513** [comment]
I tried "npm install appium" and i also tried installing 1.3.6 through cmd using "npm install [-g] appium@1.3.6" but both ways i am getting error " Error: ENOENT, stat 'C:\Users\mshivak\AppData\Roaming\npm' "

**Comment by 0x1mason** [comment]
The brackets mean the *-g* is optional.  Try `mkdir C:\Users\mshivak\AppData\Roaming\npm`. Then run `npm install appium@1.3.6`.

**Comment by vir513** [comment]
tried it. It gives fetch failed error while downloading:

npm ERR! fetch failed https://registry.npmjs.org/mv/-/mv-2.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/npmlog/-/npmlog-0.1.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/glob/-/glob-3.1.21.tgz
npm ERR! fetch failed https://registry.npmjs.org/grunt-legacy-util/-/grunt-legac
y-util-0.2.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/buffer-crc32/-/buffer-crc32-0.2
.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore.string/-/underscore.
string-2.4.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore.string/-/underscore.
string-2.2.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/es6-mapify/-/es6-mapify-1.0.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.7.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/iconv-lite/-/iconv-lite-0.2.11.
tgz
npm ERR! fetch failed https://registry.npmjs.org/sax/-/sax-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/rtts-assert/-/rtts-assert-0.0.7
.tgz
npm ERR! fetch failed https://registry.npmjs.org/request/-/request-2.16.6.tgz
npm ERR! fetch failed https://registry.npmjs.org/cycle/-/cycle-1.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/eyes/-/eyes-0.1.8.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.4.4.t
gz
npm ERR! fetch failed https://registry.npmjs.org/optimist/-/optimist-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/winston/-/winston-0.6.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/graceful-fs/-/graceful-fs-3.0.6
.tgz
npm ERR! fetch failed https://registry.npmjs.org/once/-/once-1.3.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/inherits/-/inherits-2.0.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/base64-js/-/base64-js-0.0.6.tgz

npm ERR! fetch failed https://registry.npmjs.org/minimatch/-/minimatch-1.0.0.tgz

npm ERR! fetch failed https://registry.npmjs.org/util-deprecate/-/util-deprecate
-1.0.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/uglify-js/-/uglify-js-2.4.17.tg
z
npm ERR! fetch failed https://registry.npmjs.org/underscore.string/-/underscore.
string-2.2.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/es6-mapify/-/es6-mapify-1.0.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.7.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/iconv-lite/-/iconv-lite-0.2.11.
tgz
npm ERR! fetch failed https://registry.npmjs.org/sax/-/sax-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/rtts-assert/-/rtts-assert-0.0.7
.tgz
npm ERR! fetch failed https://registry.npmjs.org/request/-/request-2.16.6.tgz
npm ERR! fetch failed https://registry.npmjs.org/cycle/-/cycle-1.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/eyes/-/eyes-0.1.8.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.4.4.t
gz
npm ERR! fetch failed https://registry.npmjs.org/optimist/-/optimist-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/winston/-/winston-0.6.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/graceful-fs/-/graceful-fs-3.0.6
.tgz
npm ERR! fetch failed https://registry.npmjs.org/once/-/once-1.3.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/inherits/-/inherits-2.0.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/minimatch/-/minimatch-1.0.0.tgz

npm ERR! fetch failed https://registry.npmjs.org/base64-js/-/base64-js-0.0.6.tgz

npm ERR! fetch failed https://registry.npmjs.org/uglify-js/-/uglify-js-2.4.17.tg
z
npm ERR! fetch failed https://registry.npmjs.org/util-deprecate/-/util-deprecate
-1.0.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore.string/-/underscore.
string-2.2.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/es6-mapify/-/es6-mapify-1.0.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.7.0.t
gz
npm ERR! fetch failed https://registry.npmjs.org/iconv-lite/-/iconv-lite-0.2.11.
tgz
npm ERR! fetch failed https://registry.npmjs.org/sax/-/sax-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/rtts-assert/-/rtts-assert-0.0.7
.tgz
npm ERR! fetch failed https://registry.npmjs.org/request/-/request-2.16.6.tgz
npm ERR! fetch failed https://registry.npmjs.org/cycle/-/cycle-1.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/eyes/-/eyes-0.1.8.tgz
npm ERR! fetch failed https://registry.npmjs.org/underscore/-/underscore-1.4.4.t
gz
npm ERR! fetch failed https://registry.npmjs.org/optimist/-/optimist-0.6.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/winston/-/winston-0.6.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/graceful-fs/-/graceful-fs-3.0.6
.tgz
npm ERR! fetch failed https://registry.npmjs.org/inherits/-/inherits-2.0.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/once/-/once-1.3.1.tgz
npm ERR! fetch failed https://registry.npmjs.org/minimatch/-/minimatch-1.0.0.tgz

npm ERR! fetch failed https://registry.npmjs.org/base64-js/-/base64-js-0.0.6.tgz

npm ERR! fetch failed https://registry.npmjs.org/uglify-js/-/uglify-js-2.4.17.tg
z
npm ERR! fetch failed https://registry.npmjs.org/util-deprecate/-/util-deprecate
-1.0.0.tgz
npm WARN optional dep failed, continuing form-data@0.1.4
npm WARN optional dep failed, continuing http-signature@0.10.1
npm WARN optional dep failed, continuing tough-cookie@0.12.1
npm WARN optional dep failed, continuing hawk@1.0.0
npm ERR! fetch failed https://registry.npmjs.org/ws/-/ws-0.4.31.tgz
npm ERR! fetch failed https://registry.npmjs.org/socket.io-parser/-/socket.io-pa
rser-2.1.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/debug/-/debug-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/send/-/send-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/parseurl/-/parseurl-1.1.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/appium-support/-/appium-support
-0.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/ncp/-/ncp-0.4.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/socket.io-parser/-/socket.io-pa
rser-2.1.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/debug/-/debug-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/ws/-/ws-0.4.31.tgz
npm ERR! fetch failed https://registry.npmjs.org/parseurl/-/parseurl-1.1.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/send/-/send-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/appium-support/-/appium-support
-0.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/ncp/-/ncp-0.4.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/socket.io-parser/-/socket.io-pa
rser-2.1.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/debug/-/debug-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/ws/-/ws-0.4.31.tgz
npm ERR! fetch failed https://registry.npmjs.org/parseurl/-/parseurl-1.1.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/send/-/send-0.6.0.tgz
npm ERR! fetch failed https://registry.npmjs.org/ncp/-/ncp-0.4.2.tgz
npm ERR! fetch failed https://registry.npmjs.org/appium-support/-/appium-support
-0.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/eyes/-/eyes-0.1.8.tgz
npm ERR! fetch failed https://registry.npmjs.org/cycle/-/cycle-1.0.3.tgz
npm ERR! fetch failed https://registry.npmjs.org/eyes/-/eyes-0.1.8.tgz
npm ERR! fetch failed https://registry.npmjs.org/cycle/-/cycle-1.0.3.tgz

and goes on...

**Comment by frankyliu** [comment]
maybe you can find some local mirror for the npm repository

**Comment by 0x1mason** [comment]
Closing this. Samsung devices don't have the Browser app. They have a specialized Samsung version which doesn't play nice with Chrome. so it's a wontfix.

**Comment by misskhushboo** [comment]
What is the roadmap for the support of Native browser automation on Samsung, HTC devcies?

**Comment by jlipps** [comment]
It's not on the roadmap, since Samsung doesn't use the Chromium webview so we can't automate it with our usual strategy.

**Comment by misskhushboo** [comment]
Please recommened on which device can I automate the native android browser, I tried Samsung and Htc and failed to do, Nexus is not having the native browser app?

**Comment by jlipps** [comment]
It works on the Google devices, but those come with Chrome pre-installed so you don't even need to automate the Browser app there.
